test/python/test_qasm_parser.py

```diff
@@ -78,5 +81,37 @@ def test_get_tokens(self):
             self.assertTrue(isinstance(token, ply.lex.LexToken))
 
 
+class TestParserWithComments(QiskitTestCase):
+    """QasmParser has a with_comment option to get have comments as extra nodes"""
+
+    def test_single_comment(self):
+        """The thing to parse is a single comment"""
+        qasm_string = """//this is a comment"""
+        ast = Qasm(data=qasm_string).parse(with_comments=True)
+        self.assertEqual(len(ast.children), 1)
+        self.assertIsInstance(ast.children[0], Comment)
+        self.assertEqual(ast.qasm().replace('\n', '', 1), qasm_string)
```

Comment:
Non-blocking: I'm against modifying the value I want to test, I prefer to modify the other one or provide an exact representation. Don't you think?

Reply:
I agree with the principle. The "unparser" has some issues. More details in #901.

Reply:
The strings are short right now; we could, at least, write the expected string explicitely so I have not to imagine how it should look like. Is it worth the effort?

Reply:
Sure ee5afc8